java/code/src/com/suse/manager/webui/controllers/admin/handlers/HubApiController.java

```diff
@@ -146,6 +162,104 @@ private String removeHubRootCA(Request request, Response response, User user) {
         return updateServerRootCA(request, response, user, IssRole.HUB, null);
     }
 
+    private String getPeripheralChannelSyncStatus(Request request, Response response, User satAdmin) {
+        long peripheralId = Long.parseLong(request.params("id"));
+        try {
+            ChannelSyncModel syncModel = hubManager.getChannelSyncModelForPeripheral(satAdmin, peripheralId);
+            return json(response, GSON.toJson(syncModel));
+        }
+        catch (CertificateException eIn) {
+            LOGGER.error("Unable to parse the specified root certificate for the peripheral {}", peripheralId, eIn);
+            return badRequest(response, LOC.getMessage("hub.invalid_root_ca"));
+        }
+        catch (IOException eIn) {
+            LOGGER.error("Error while attempting to connect to peripheral server {}", peripheralId, eIn);
+            return internalServerError(response, LOC.getMessage("hub.error_connecting_remote"));
+        }
+    }
+
+    /**
+     * Model for sync channel operations
+     */
+    private static class SyncChannelModel {
```

Comment:
I would move this class to its own file, in the `com.suse.manager.webui.controllers.admin.beans` package like the others we have already